File: reduce.py
#OfTheAbsurd (Donia Tung & Samantha Ngo)
# SoftDev2 pd7
# K18 -- Reductio ad Absurdum
# 2018-04-30

from functools import reduce

# Read in contents -------------------------------------------------------------------------------------------------------------------------------------------
story = "alice.txt"
f = open(story, 'r')

contents = f.read().split();
contents = [x.strip('\n') for x in contents]
contents = [x.strip(',') for x in contents]

#print contents

# Count frequency of a single word ---------------------------------------------------------------------------------------------------------------------------
def counter(word, file):
    list = [1 for x in file if x == word]
    if list == []:
        return 0
    return reduce ((lambda x, y: x + y), list)

test = "hi there this here sentence is the test sentence so it is here so that there is a sentence for testing".split()

print counter("is", test) #3
print counter("here", test) #2

# Find frequency of a group of words(phrase) ----------------------------------------------------------------------------------------------------------------
def frequencyOfGroup(word_list):
    freq = 0;
    print "GROUP FREQ: ", [counter(x, contents) for x in word_list]

print frequencyOfGroup(['the','Alice'])
print "=============================="

def addedGroupFreq(word_list):
    freq = [counter(x, contents) for x in word_list]
    print "GROUP FREQ: ", freq
    return reduce((lambda x,y : x+y), freq)

print addedGroupFreq(['the','Alice'])
print "=============================="

# Find most frequent word (takes a long time) ----------------------------------------------------------------------------------------------------------------
def countGroup(group, file):
    return true

def mostFrequent(file):
    wordfreq = [(counter(x, file), x) for x in file]
    return "Most Frequent Word: '" + max(wordfreq)[1] + "' appears " + str(max(wordfreq)[0]) + " times."

print mostFrequent(test)
print "TEST BOOK:"
#print mostFrequent(contents)

File: reduce.js

var test = "hi there this here sentence is the test sentence so it is here so that there is a sentence for testing".split(" ")
//console.log(test)

var x = [1, 2, 3, 4]
var a = x.reduce(function(a, b) {return a + b})

//console.log(a);

//in Python...
/*def counter(word, file):
    list = [1 for x in file if x == word]
    if list == []:
        return 0
    return reduce ((lambda x, y: x + y), list)*/

//in Javascript...
var frequency = function(word, file){
  var total = [];
  for (x in file){
    if (file[x] == word){
      total.push(1);
    }
  }
  if (total.length == 0){
      return 0;
  }
  return total.reduce(function (x, y) {return x + y})
}
//console.log(test)
console.log(frequency("sentence", test))

/*In Python...
def frequencyOfGroup(word_list):
    freq = 0;
    print "GROUP FREQ: ", [counter(x, contents) for x in word_list]
*/

//In JS...
var freqOfPhrase = function(word_list, file){
  var freqCount = []
  for (x in word_list){
    freqCount.push(frequency(word_list[x], file))
  }
  return freqCount.reduce(function(x,y) {return x + y})
}

console.log(freqOfPhrase(["is", "sentence"], test))

/*In Python...
def mostFrequent(file):
    wordfreq = [(counter(x, file), x) for x in file]
    return "Most Frequent Word: '" + max(wordfreq)[1] + "' appears " + str(max(wordfreq)[0]) + " times."
*/

//In JS...
var mostFrequent = function(file){
  var currentFreq = [];
  for (x in file){
    console.log(file[x])
    currentFreq.push(frequency(file[x]), file);
  }
  var index = Math.max(currentFreq)
  return "Word: " + file[index] + " frequency: " + currentFreq[index];
}

console.log(mostFrequent(test))
